libs/langchain/langchain/llms/fireworks.py

```diff
@@ -66,43 +70,89 @@ def _llm_type(self) -> str:
         """Return type of llm."""
         return "fireworks"
 
-    def _call(
+    def _generate(
         self,
-        prompt: str,
+        prompts: List[str],
         stop: Optional[List[str]] = None,
         run_manager: Optional[CallbackManagerForLLMRun] = None,
         **kwargs: Any,
-    ) -> str:
-        """Run the LLM on the given prompt and input."""
-        params: dict = {
+    ) -> LLMResult:
+        """Call out to Fireworks endpoint with k unique prompts.
+        Args:
+            prompts: The prompts to pass into the model.
+            stop: Optional list of stop words to use when generating.
+        Returns:
+            The full LLM output.
+        """
+        params = {
             "model": self.model,
-            "prompt": prompt,
             **self.model_kwargs,
         }
-        response = completion_with_retry(
-            self, run_manager=run_manager, stop=stop, **params
-        )
+        sub_prompts = self.get_batch_prompts(params, prompts, stop)
+        choices = []
+        for _prompts in sub_prompts:
+            response = completion_with_retry_batching(
```

Comment:
need to pass stop words in here, I think

Reply:
`stop` is in params.

Reply:
See other note, I don't believe this is correct... please double check.

Reply:
Are you sure? maybe I am reading this wrong but I think that is not the case... params only contains **self.model_kwargs not the **kwargs for this function. Even if you passed in the **kwargs for this function, that would not include stop since that is a named parameter.